perl/lib/BarnOwl/DeferredLogging.pm

```diff
@@ -0,0 +1,61 @@
+use strict;
+use warnings;
+
+package BarnOwl::DeferredLogging;
+
+=head1 BarnOwl::DeferredLogging
+
+=head1 DESCRIPTION
+
+C<BarnOwl::DeferredLogging> adds variables relevant to deferred logging.
+
+=head2 USAGE
+
+Not Applicable.
+
+=head2 EXPORTS
+
+None by default.
+
+=cut
+
+use BarnOwl::Timer;
+use Exporter;
+
+our @EXPORT_OK = qw();
+
+our %EXPORT_TAGS = (all => [@EXPORT_OK]);
+
+$BarnOwl::Hooks::startup->add("BarnOwl::DeferredLogging::_register_variables");
+
+sub _register_variables {
+    my $flush_logs_interval = -1;
+    my $flush_logs_timer;
+    BarnOwl::new_variable_full('flush-logs-interval',
+        {
+            default        => -1,
```

Comment:
I kinda wish we we could have a positive default (60 minutes? 360?), but I'm not sure how I feel about the "drop anything that fails for not-EACCES/EPERM/ETIMEDOUT" rule. I guess the prior rule was "drop anything that fails", so not so bad...?